net/base/directory_lister.h

```diff
@@ -48,6 +45,40 @@ class NET_EXPORT DirectoryLister  {
     virtual ~DirectoryListerDelegate() {}
   };
 
+  enum ListingType {};
+
+  DirectoryLister(const base::FilePath& dir,
+                  DirectoryListerDelegate* delegate){};
+
+  DirectoryLister(const base::FilePath& dir,
+                  ListingType type,
+                  DirectoryListerDelegate* delegate){};
+
+  // Will invoke Cancel().
+  ~DirectoryLister(){};
+
+  // Call this method to start the asynchronous directory enumeration.
+  void Start() {};
+
+  // Call this method to asynchronously stop directory enumeration.  The
+  // delegate will not be called back.
+  void Cancel() {};
+};
+
+#else
+
+class NET_EXPORT DirectoryLister  {
+ public:
+  // Represents one file found.
+  struct DirectoryListerData {
```

Comment:
it was kaido's idea to stuvb out the entire class rather than what's needed : that just makes it more cleaner in my opinion , rather have having it ifdef in multiple multiple place - my understnading is this will help when we update base and net

Reply:
The cleanest path would be still to remove the call site, or the place where this entire thing gets referenced from. If it's just 1 caller, then it's probably better to stub it out from caller side. If it's multiple for some reason, putting as minimal stub/symbol in here with a clear comment as to why it exists is probably best.

Reply:
Multiple functions are referenced from multiple files - whatever has been referenced has been stubbed out.

Note that directory_lister.h was not present in the gn files list before this so stubbing out the class seems pretty safe.

I'm closing this conversation , feel free to reopen if you want me to add more info apart from the bug i added for the entire class // TODO b/296715826 Fix stubbed out class.